tests/test_main.py

```diff
@@ -601,6 +609,48 @@ def test_enable_and_disable_mutation_type_are_exclusive():
     assert "You can't combine --disable-mutation-types and --enable-mutation-types" in result.output
 
 
+@pytest.mark.parametrize(
+    "mutation_type, expected_mutation",
+    [
+        ("expr_stmt", "result = None"),
+        ("operator", "result = a - b"),
+    ]
+)
+def test_show_mutant_after_run_with_disabled_mutation_types(surviving_mutants_filesystem, mutation_type, expected_mutation):
+    """Test for issue #234: ``mutmut show <id>`` did not show the correct mutant if ``mutmut run`` was
+    run with ``--enable-mutation-types`` or ``--disable-mutation-types``."""
+    CliRunner().invoke(climain, ['run', '--paths-to-mutate=foo.py', f'--enable-mutation-types={mutation_type}'], catch_exceptions=False)
+    result = CliRunner().invoke(climain, ['show', '1'])
+    assert f"""
+ def foo(a, b):
+-    result = a + b
++    {expected_mutation}
+     return result
+""" in result.output
+
+
+def test_run_multiple_times_with_different_mutation_types(filesystem):
+    """Running multiple times with different mutation types enabled should append the new mutants to the cache without
+    altering existing mutants."""
```

Comment:
I wondered if running different mutation types one after the other works without breaking the cache. Luckily it does!